### PR TITLE
feat: Make the default authHttpHandler in CredentialsWrapper null

### PR DESCRIPTION
**Pull request by saranshdhingra**
Currently the [v2 samples for Pubsub](https://github.com/GoogleCloudPlatform/php-docs-samples/pull/1967) are failing due to an internal attempt to `serialize` the `PubSubClient` in the `BatchPublisher`.
This is because the default `authHttpHandler` contains a closure. With this PR we make the default handler as `null`.

### CODE REVIEW

**Review by yash30201 — APPROVED** [review]
LGTM.

This makes sense as `ApplicationDefaultCredentials` in auth library creates its own httphandlerif null is supplied.

File: src/CredentialsWrapper.php [hunk 1]
```diff
@@ -141,7 +141,7 @@ public static function build(
         ];
 
         $keyFile = $args['keyFile'];
-        $authHttpHandler = $args['authHttpHandler'] ?: self::buildHttpHandlerFactory();
+        $authHttpHandler = $args['authHttpHandler'];
```
Comment (bshaffer) [inline src/CredentialsWrapper.php:144]: nit: let's simplify this and just make the $authHttpHandler supplied to the loader and `CredentialWrapper` come directly from `$args`

File: tests/Tests/Unit/CredentialsWrapperTest.php [hunk 2]
```diff
@@ -83,7 +83,7 @@ public function buildDataWithoutExplicitKeyFile()
         $appDefaultCreds = getenv('GOOGLE_APPLICATION_CREDENTIALS');
         putenv('GOOGLE_APPLICATION_CREDENTIALS=' . __DIR__ . '/testdata/json-key-file.json');
         $scopes = ['myscope'];
-        $defaultAuthHttpHandler = HttpHandlerFactory::build();
+        $defaultAuthHttpHandler = null;
```
Comment (bshaffer) [inline tests/Tests/Unit/CredentialsWrapperTest.php:86]: It defaults to `null`, so why pass this in at all? Seems better to get rid of it and all instances of it being passed in to simplify things.

**Review by bshaffer — COMMENTED** [review]
Let's add a test that verifies the `CredentialsWrapper` class is able to be serialized by default:

```php
public function testSerializeCredentialsWrapper()
{
    $credentialsWrapper = CredentialsWrapper::build(['credentials' => new ServiceAccoutnCredentials($testJson)]);
    $foo = serialize($credentialsWrapper);
    $this->assertIsString($foo);
}
```

**Review by bshaffer — APPROVED** [review]

**Pull request merged**